test/test_server.c

```diff
@@ -3358,6 +3358,130 @@ void test_0x3e_suppress_positive_response(void **state) {
     TEST_INT_EQUAL(UDSTpRecv(e->client_tp, buf, sizeof(buf), NULL), 0);
 }
 
+UDSErr_t fn_test_0x85_control_dtc_setting(UDSServer_t *srv, UDSEvent_t ev, void *arg) {
+    TEST_INT_EQUAL(ev, UDS_EVT_ControlDTCSetting);
+
+    UDSControlDTCSettingArgs_t *args = arg;
+
+    switch (args->type) {
+    case 0x01:
+        return UDS_PositiveResponse;
+    case 0x02:
+        TEST_INT_EQUAL(args->len, 2)
+        uint8_t expected_data[] = {0xF1, 0xF2};
+        TEST_MEMORY_EQUAL(args->data, expected_data, sizeof(expected_data))
+        return UDS_PositiveResponse;
+    case 0x40:
+        return UDS_NRC_ConditionsNotCorrect;
+    }
+
+    return UDS_NRC_GeneralProgrammingFailure;
+}
+
+void test_0x85_control_dtc_setting(void **state) {
```

Comment:
nit: if the test case implements a message flow defined in one of the standards, please comment it as such. This looks like ISO14229-1:2020(E) 10.8.5.2.

The idea here is that test cases described by the standards are already well documented. If the test case does not exist in the standard, it merits additional documentation as comments, link to an issue, etc.